.circleci/config.yml

```diff
@@ -0,0 +1,73 @@
+version: 2.1
+
+default: &default
+  steps:
+    - checkout
+    - run: git submodule sync
+    - run: git submodule update --init
+    - run: pip install --user tox
+    - run: scripts/circle/install_node.sh
+    - run:
+        name: Add node to the path
+        command: |
+          echo 'export PATH=~/.nvm/versions/node/v${NODE_VERSION}/bin:$PATH' >> $BASH_ENV
+          source $BASH_ENV
+    - run: tox
+
+jobs:
+  tests:
+    <<: *default
+    docker:
+      - image: 'cimg/python:3.6'
+        environment:
+          TOXENV: py36,codecov
+          TOX_POSARGS: ''
+      - image: 'docker.elastic.co/elasticsearch/elasticsearch:6.8.12'
+        name: search
+
+  lint:
+    <<: *default
+    docker:
+      - image: 'cimg/python:3.6'
+        environment:
+          TOXENV: lint
+
+  migrations:
+    <<: *default
+    docker:
+      - image: 'cimg/python:3.6'
+        environment:
+          TOXENV: migrations
+
+  docs:
+    <<: *default
+    docker:
+      - image: 'cimg/python:3.6'
+        environment:
+          TOXENV: docs
+
+  docs-lint:
+    <<: *default
+    docker:
+      - image: 'cimg/python:3.6'
+        environment:
+          TOXENV: docs-lint
+
+  eslint:
+    <<: *default
+    docker:
+      - image: 'cimg/python:3.6'
+        environment:
+          TOXENV: eslint
+          NODE_VERSION: 10.17.0
```

Comment:
Looks like we can speed up things by running all these in one job, otherwise circle will create a new container per each job and we can only have 5 concurrent jobs

Reply:
I don't fully follow. Meaning just have 1 job that calls `tox` and it runs all 5 steps? I think we do that on corporate.

Reply:
yeah, just call `tox` with all except for the py36 environment (since that requires ES running, but maybe isn't that bad having ES running during all tests?). Also, splitting will give us the linter failures more quickly :D

Reply:
I think 1 job per tox step is easier to parse while reading the errors. Right now we have 6 check notifications in this PR (1 per job) and you know immediately _what_ (docs, lint, etc) failed without going to CircleCI.

Also, if we have 1 job per tox step, don't they run in parallel?

Reply:
> Also, if we have 1 job per tox step, don't they run in parallel?

Yeah, but then circle will be creating one container per each task, and with our current plan we can only have 5 concurrent jobs. So other PRs will need to wait till there is space available.

Reply:
I think we can run `tox -e <env>` on separate steps so is more clear what failed